beaconclient/prod_beacon_instance.go

```diff
@@ -240,25 +240,27 @@ type GetBlockResponse struct {
 func (c *ProdBeaconInstance) GetBlock(blockID string) (block *GetBlockResponse, err error) {
 	uri := fmt.Sprintf("%s/eth/v2/beacon/blocks/%s", c.beaconURI, blockID)
 	resp := new(GetBlockResponse)
-	_, err = fetchBeacon(http.MethodGet, uri, nil, resp, nil)
+	_, err = fetchBeacon(http.MethodGet, uri, nil, resp, nil, http.Header{})
 	return resp, err
 }
 
 // GetBlockForSlot returns the block for a given slot - https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2
 func (c *ProdBeaconInstance) GetBlockForSlot(slot uint64) (*GetBlockResponse, error) {
 	uri := fmt.Sprintf("%s/eth/v2/beacon/blocks/%d", c.beaconURI, slot)
 	resp := new(GetBlockResponse)
-	_, err := fetchBeacon(http.MethodGet, uri, nil, resp, nil)
+	_, err := fetchBeacon(http.MethodGet, uri, nil, resp, nil, http.Header{})
 	return resp, err
 }
 
 func (c *ProdBeaconInstance) GetURI() string {
 	return c.beaconURI
 }
 
-func (c *ProdBeaconInstance) PublishBlock(block *common.SignedBeaconBlock) (code int, err error) {
-	uri := fmt.Sprintf("%s/eth/v1/beacon/blocks", c.beaconURI)
-	return fetchBeacon(http.MethodPost, uri, block, nil, nil)
+func (c *ProdBeaconInstance) PublishBlock(block *common.SignedBeaconBlock, broadcastValidation BroadcastValidation) (code int, err error) {
+	uri := fmt.Sprintf("%s/eth/v2/beacon/blocks?broadcast_validation=%s", c.beaconURI, broadcastValidation.String())
+	headers := http.Header{}
+	headers.Add("Eth-Consensus-Version", common.ForkVersionStringCapella)
```

Comment:
Will this work with deneb? Could we un-hardcode this?

Reply:
yes, as part of #489 planning to switch over to a `VersionedSignedBeaconBlock` in order to get the consensus version